### PR TITLE
Backport commits and update go module dependencies

### PR DESCRIPTION
**Pull request by masnax**
The `v1` branch was set up at the commit currently used by MicroCloud `1.1` but there are some additional commits prior to us introducing breaking changes that can be backported.

Those, as well as an updated set of dependencies, are included here.

### CODE REVIEW

**Comment by masnax** [comment]
@simondeziel FYI, Seems at some point you [merged a PR](https://github.com/canonical/microcluster/pull/302/commits/ee9ec1705da5ea2cd0921556c20f83e5678dd1bf) with your gmail account, so I swapped the author field with your canonical email to get past DCO.

**Comment by simondeziel** [comment]
> @simondeziel FYI, Seems at some point you [merged a PR](https://github.com/canonical/microcluster/pull/302/commits/ee9ec1705da5ea2cd0921556c20f83e5678dd1bf) with your gmail account, so I swapped the author field with your canonical email to get past DCO.

Sorry about that. I've now made some changes to my account setting to hopefully avoid any such mistake in the future. Thanks for letting me know!

**Comment by masnax** [comment]
In the end I decided to clear out some of the more unnecessary backports and instead focus on just bugfixes and go module updates.

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", "*" ]
```
Comment (roosterfish) [inline .github/workflows/codeql.yml:16]: I guess this commit isn't required as the workflow anyway is only running on `main`, but it doesn't hurt either.

File: go.mod [hunk 2]
```diff
@@ -1,53 +1,55 @@
...
-	golang.org/x/sys v0.18.0
+	github.com/spf13/cobra v1.8.1
+	golang.org/x/sys v0.28.0
 	gopkg.in/yaml.v2 v2.4.0
 )
 
 require (
 	github.com/Rican7/retry v0.3.1 // indirect
 	github.com/armon/go-proxyproto v0.1.0 // indirect
+	github.com/canonical/go-dqlite/v3 v3.0.0 // indirect
```
Comment (roosterfish) [inline go.mod]: The go-dqlite/v3 looks suspicious. Shouldn't this be v2 only? I am wondering from where it's coming from.

**Review by roosterfish — COMMENTED** [review]
Only a small comment on the pulled indirect deps.

File: go.mod (hunk 2 above)
Comment (masnax) [reply go.mod]: Ah I missed a `go mod tidy` in the makefile.

**Review by roosterfish — APPROVED** [review]
LGTM!

**Pull request merged**